shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLUTTERPLUGIN_H_
+#define FLUTTER_FLUTTERPLUGIN_H_
+
+#include "FlutterBinaryMessenger.h"
+#include "FlutterChannels.h"
+#include "FlutterCodecs.h"
+
+NS_ASSUME_NONNULL_BEGIN
+@protocol FlutterPluginRegistrar;
+
+@protocol FlutterPlugin <NSObject>
+@required
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar;
+@optional
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result;
+- (BOOL)application:(UIApplication*)application didFinishLaunchingWithOptions:(NSDictionary*)launchOptions;
```

Comment:
Almost, but not quite. Because some of these callbacks should logically be handled by only one plugin, I declare them with a `BOOL` return type and only iterate until the first plugin answers "`YES`, I handled it".

BTW: looking at https://github.com/flutter/firebase_messaging it seems possible to remove the `UIApplication` argument from all of these signatures, because the plugin can alswas do `[UIApplication sharedApplication]` to access it, if needed.

What do you think?